router/batchrouter/handle_async.go

```diff
@@ -32,300 +194,29 @@ func (brt *Handle) pollAsyncStatus(ctx context.Context) {
 			destinationsMap := brt.destinationsMap
 			brt.configSubscriberMu.RUnlock()
 
-			for key := range destinationsMap {
+			for destinationID := range destinationsMap {
 				if slices.Contains(asyncDestinations, brt.destType) {
```

Comment:
```suggestion
			for destinationID := range destinationsMap {
```
We can get rid of the `slices.Contains` here and ensure that we spin the `pollAsyncStatus` only for async destinations?

Reply:
@koladilip

Reply:
addressed